modules/dnn/src/opencl/math.cl

```diff
@@ -0,0 +1,155 @@
+__global Dtype* Y,
+                                         const int offY) {
+  for (int index = get_global_id(0); index < N; index += get_global_size(0)) {
+    Y[offY + index] += alpha;
+  }
+}
+
+__kernel void TEMPLATE(add,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa, __global const Dtype* b,
+                                  const int offb, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[offy + index] = a[offa + index] + b[offb + index];
+  }
+}
+
+__kernel void TEMPLATE(sub,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa, __global const Dtype* b,
+                                  const int offb, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[offy + index] = a[offa + index] - b[offb + index];
+  }
+}
+
+__kernel void TEMPLATE(abs,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[offy + index] = fabs((Dtype)(a[offa + index]));
+  }
+}
+
+__kernel void TEMPLATE(exp,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[offy + index] = exp(a[offa + index]);
+  }
+}
+
+__kernel void TEMPLATE(log,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[offy + index] = log((Dtype)(a[offa + index]));
+  }
+}
+
+__kernel void TEMPLATE(powx,Dtype)(const int n, __global const Dtype* a,
+                                   const int offa, Dtype alpha,
+                                   __global Dtype* y,
+                                   const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    if(alpha == 2.0) {
+      y[offy + index] = pow((Dtype)fabs(a[offa + index]), (Dtype)alpha);
+    } else {
+      y[offy + index] = pow((Dtype)a[offa + index], (Dtype)alpha);
+    }
+  }
+}
+
+__kernel void TEMPLATE(sign,Dtype)(const int n, __global const Dtype* x,
+                                   const int offx, __global Dtype* y,
+                                   const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[index + offy] = (0.0 < x[index + offx])
+        - (x[index + offx] < 0.0);
```

Comment:
What is about [sign()](https://www.khronos.org/registry/OpenCL/sdk/1.2/docs/man/xhtml/sign.html) ?